auth_api_key_hash/models/auth_api_key_hash.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Solvti Sp. z o.o.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from hashlib import md5
+
+from odoo import _, api, fields, models, tools
+from odoo.exceptions import AccessError, ValidationError
+from odoo.tools import consteq
+
+
+class AuthApiKeyHash(models.Model):
+    _name = "auth.api.key.hash"
+    _description = "API Key (Hash)"
+
+    name = fields.Char(required=True)
+    key = fields.Char(required=True, copy=False, help="""The API key.""",)
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="User",
+        required=True,
+        help="""The user used to process the requests authenticated by
+        the api key""",
+    )
+
+    _sql_constraints = [("name_uniq", "unique(name)", "Api Key name must be unique.")]
```

Comment:
Maybe unique on `key`, `user_id` pair would be a good idea as well?

Reply:
Nah, just key. It is better. The key must be definitely unique and not repeated.

Reply:
added key uniqe